File: dialog/src/main/java/com/github/lany192/dialog/LoadingDialog.java
package com.github.lany192.dialog;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.text.TextUtils;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.StringRes;

import com.github.lany192.dialog.databinding.DialogLoadingBinding;

import java.util.Objects;

/**
 * 加载对话框
 */
public class LoadingDialog extends BaseDialog<DialogLoadingBinding> {
    private CharSequence mMessage;

    @Override
    protected void init() {
        if (!TextUtils.isEmpty(mMessage)) {
            binding.message.setText(mMessage);
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        Window window = Objects.requireNonNull(getDialog()).getWindow();
        if (window != null) {
            window.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
            window.setLayout(WindowManager.LayoutParams.WRAP_CONTENT, WindowManager.LayoutParams.WRAP_CONTENT);
            WindowManager.LayoutParams layoutParams = window.getAttributes();
            layoutParams.dimAmount = 0.0f;//设置背景全透明
            window.setAttributes(layoutParams);
        }
    }

    public void setMessage(CharSequence message) {
        this.mMessage = message;
    }

    public void setMessage(@StringRes int resId) {
        this.mMessage = getString(resId);
    }
}

File: library/dialog/src/main/java/com/github/lany192/dialog/LoadingDialog.kt
package com.github.lany192.dialog

import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.text.TextUtils
import android.view.WindowManager
import androidx.annotation.StringRes
import com.github.lany192.dialog.databinding.DialogLoadingBinding
import com.github.lany192.utils.ContextUtils

/**
 * 加载对话框
 */
class LoadingDialog : BaseDialog<DialogLoadingBinding>() {

    private var mMessage: CharSequence? = null

    override fun init() {
        if (!TextUtils.isEmpty(mMessage)) {
            binding.message.text = mMessage
        }
    }

    override fun onResume() {
        super.onResume()
        dialog?.window?.let {
            it.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            it.setLayout(
                WindowManager.LayoutParams.WRAP_CONTENT,
                WindowManager.LayoutParams.WRAP_CONTENT
            )
            val layoutParams = it.attributes
            layoutParams.dimAmount = 0.0f //设置背景全透明
            it.attributes = layoutParams
        }
    }

    fun setMessage(message: CharSequence?) {
        mMessage = message
    }

    fun setMessage(@StringRes resId: Int) {
        mMessage = ContextUtils.getContext().getString(resId)
    }
}